flux_local/values.py

```diff
@@ -173,6 +173,19 @@ def _lookup_value_reference(
 
     return found_value
 
+def deep_merge(base: dict, override: dict) -> dict:
+    """
+    Recursively merge two dictionaries, similar to how Helm merges values.
+    Lists are replaced entirely (Helm behavior).
+    """
+    result = base.copy()
+    for key, value in override.items():
+        if key in result and isinstance(result[key], dict) and isinstance(value, dict):
```

Comment:
Extract `result.get(key)` into a variable since it is looked up 3 times:

Reply:
Extracted into a variable